### PR TITLE
Fixed width dashboard migration

### PR DESCRIPTION
**Pull request by adam-james-v**
This PR addressed the first 2 items in #36358

> - [x] (BE) Create new width setting on dashboard object with 2 options: "full" | "fixed"
> - [x] (BE) Create migration to set all existing dashboards to width = "full"

The need was:
 - add a :width setting on dashboards
 - the allowed values here are 'fixed' or 'full'
 - for all new dashboards, the default setting shall be 'fixed' (the new default behaviour we want)
 - for all existing dashboards, the value should be set to 'full' (the existing behaviour)

3 migrations were used to achieve this:
 - the migration adding the column, whose default value is 'fixed'. This ensures all new dashboards will have the value correctly set
 - another migration to change all existing dashboard entries to have the 'full' value. This makes sense as it won't break existing behaviours for users
 - the final migration setting the width column to notNull, ensuring it's always set going forward. 

A test was added to the Dashboard api test namespace to make sure these properties are met after the migration.

I've also manually tested in the REPL:

```clojure
;; an existing dashboard:
(:width (t2/select-one :model/Dashboard))
;; full.

;; a newly created dashboard:
(:width (t2/select-one :model/Dashboard :id 535))
;; fixed.

```

The setting is exposed to change via `PUT /api/dashboard/:id`.

### CODE REVIEW

File: resources/migrations/001_update_migrations.yaml [hunk 1]
```diff
@@ -5228,6 +5228,45 @@ databaseChangeLog:
...
+            columns:
+              - column:
+                  name: width
+                  value: 'full'
+      rollback: []
+
+  - changeSet:
+      id: v49.2024-01-29T19:56:40
+      author: adam-james
+      comment: Set all existing report_dashboard width values to full.
```
Comment (markbastian) [inline resources/migrations/001_update_migrations.yaml:5261]: Please update the comment and remarks to be accurate.

**Review by markbastian — CHANGES_REQUESTED** [review]
Update the comment and remarks in the last migration and then it will be g2g.

File: resources/migrations/001_update_migrations.yaml (hunk 1 above)
Comment (adam-james-v) [reply resources/migrations/001_update_migrations.yaml]: Good catch, thanks. I've updated them.

**Review by markbastian — APPROVED** [review]
Looks good.

**Review by JesseSDevaney — APPROVED** [review]

**Pull request merged**